test/exe/Main.hs

```diff
@@ -3477,6 +3485,7 @@ clientSettingsTest = testGroup "client settings handling"
             logNot <- skipManyTill anyMessage loggingNotification
             isMessagePresent "Updating Not supported" [getLogMessage logNot]
     ,   testSession "ghcide restarts shake session on config changes" $ do
+            void $ skipManyTill anyMessage $ message @RegisterCapabilityRequest
```

Comment:
Why is this required?

Reply:
Good question, i added it from pure empirical observation of existing failing tests. But i just reproduced locally the error with this version so it is not valid.
So i have to investigate and replace it.

Reply:
maybe change the test to load a file and wait for diagnostics before sending the WorkspaceDidChangeConfiguration, just to make sure that ghcide has started a build session (otherwise there will be no build session to restart)

Reply:
The session is initialized and restarted, i thought it needed a file too but then i removed it and the session logging notifications still were there (not sure why).

The log of a failing one (before waiting for ReqCaps):

<details><summary>Failing log</summary>

```
 <-- {
     "tag": "RspInitialize",
     "contents": {
.............................
     }
 }
 --> {
     "jsonrpc": "2.0",
     "params": {},
     "method": "initialized"
 }
 --> {
     "jsonrpc": "2.0",
     "params": {
         "settings": ""
     },
     "method": "workspace/didChangeConfiguration"
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": { 
             "type": 1,
             "message": "haskell-lsp:configuration parse error. NotificationMessage {_jsonrpc = \"2.0\", _method = WorkspaceDidChangeConfiguration, _params = DidChangeConfigurationParams {_settings = String \"\"}} \"Updating Not supported\""
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 1,
             "message": "haskell-lsp:configuration parse error. NotificationMessage {_jsonrpc = \"2.0\", _method = WorkspaceDidChangeConfiguration, _params = DidChangeConfigurationParams {_settings = String \"\"}} \"Updating Not supported\""
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "ReqRegisterCapability",
     "contents": {
 -----------------------------------
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 4,
             "message": "Restarting build session (aborting the previous one took 0.00s)"
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 4,
             "message": "Restarting build session (aborting the previous one took 0.00s)"
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "ReqWorkDoneProgressCreate",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "token": "1"
         },
         "method": "window/workDoneProgress/create",
         "id": 1
     }
 }
 <-- {
     "tag": "NotWorkDoneProgressBegin",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "value": {
                 "kind": "begin",
                 "title": "Processing"
             },
             "token": "1"
         },
         "method": "$/progress"
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 4,
             "message": "Finishing build session(exception: AsyncCancelled)"
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 4,
             "message": "Finishing build session(exception: AsyncCancelled)"
         },
         "method": "window/logMessage"
     }
 }
 <-- {
     "tag": "NotWorkDoneProgressEnd",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "value": {
                 "kind": "end"
             },
             "token": "1"
         },
         "method": "$/progress"
     }
 }
 <-- {
     "tag": "NotLogMessage",
     "contents": {
         "jsonrpc": "2.0",
         "params": {
             "type": 4,
             "message": "finish: kick (took 0.00s)"
         },
         "method": "window/logMessage"
     }
 }
 --> {
     "jsonrpc": "2.0",
     "params": null,
     "method": "shutdown",
     "id": 0
 }
 <-- {
     "tag": "RspShutdown",
     "contents": {
         "result": null,
         "jsonrpc": "2.0",
         "id": 0
     }
 }
 --> {
     "jsonrpc": "2.0",
     "params": [],
     "method": "exit"
 }
 FAIL
       Exception: Timed out waiting to receive a message from the server.
       Last message received:
       {
           "tag": "NotLogMessage",
           "contents": {
               "jsonrpc": "2.0",
               "params": {
                   "type": 4,
                   "message": "finish: kick (took 0.00s)"
               },
               "method": "window/logMessage"
           }
       }
```
</details>

There are one two logging notifications about `Updating Not supported` but only one `didChangeConfiguration` request. :thinking:

When the test is succesful there is only one `Updating Not Supported`:

<details> <summary> Succesful log </summary>

```
<-- {
    "tag": "RspInitialize",
    "contents": {
...................................
    }
}
--> {
    "jsonrpc": "2.0",
    "params": {},
    "method": "initialized"
}
<-- {
    "tag": "ReqRegisterCapability",
    "contents": {
.....................................
    }
}
--> {
    "jsonrpc": "2.0",
    "params": {
        "settings": ""
    },
    "method": "workspace/didChangeConfiguration"
}
<-- {
    "tag": "NotLogMessage",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "type": 1,
            "message": "haskell-lsp:configuration parse error. NotificationMessage {_jsonrpc = \"2.0\", _method = WorkspaceDidChangeConfiguration, _params = DidChangeConfigurationParams {_settings = String \"\"}} \"Updating Not supported\""
        },
        "method": "window/logMessage"
    }
}
<-- {
    "tag": "NotLogMessage",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
[DEBUG] finish: kick (took 0.00s)
            "type": 4,
            "message": "Finishing build session(exception: AsyncCancelled)"
        },
        "method": "window/logMessage"
    }
}
<-- {
    "tag": "NotLogMessage",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "type": 4,
            "message": "Restarting build session (aborting the previous one took 0.00s)"
        },
        "method": "window/logMessage"
    }
}
--> {
    "jsonrpc": "2.0",
    "params": null,
    "method": "shutdown",
    "id": 0
}
<-- {
    "tag": "ReqWorkDoneProgressCreate",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "token": "1"
        },
        "method": "window/workDoneProgress/create",
        "id": 1
    }
}
<-- {
    "tag": "NotWorkDoneProgressBegin",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "value": {
                "kind": "begin",
                "title": "Processing"
            },
            "token": "1"
        },
        "method": "$/progress"
    }
}
<-- {
    "tag": "NotWorkDoneProgressEnd",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "value": {
                "kind": "end"
            },
            "token": "1"
        },
        "method": "$/progress"
    }
}
<-- {
    "tag": "NotLogMessage",
    "contents": {
        "jsonrpc": "2.0",
        "params": {
            "type": 4,
            "message": "finish: kick (took 0.00s)"
        },
        "method": "window/logMessage"
    }
}
<-- {
    "tag": "RspShutdown",
    "contents": {
        "result": null,
        "jsonrpc": "2.0",
        "id": 0
    }
}
--> {
    "jsonrpc": "2.0",
    "params": [],
    "method": "exit"
}
```

</details>

The intent of the test is "send the config change request and parse the three first logging notifications thereafter".
In both cases there are three logging notifications so i dont know why it get stuck until the timeout.